### PR TITLE
TW-1074: [tech] Upgrade `react-native`

### PR DESCRIPTION
**Pull request by alex-tsx**
Task: https://madfish.atlassian.net/browse/TW-1074

Upgrade manual: https://react-native-community.github.io/upgrade-helper/?from=0.70.9&to=0.72.4

\+ Enabled Hermes on iOS
\+ Updated `react-native-bootsplash` by https://github.com/zoontek/react-native-bootsplash/blob/master/MIGRATION.md
\+ Updated `@taquito` & `@airgap/beacon-sdk`
\- Updated `react-native-`: `-reanimated`, `-screens`, `-safe-area-context`
\+ Completely erased E2E tests & their dependencies

Results for `yarn audit`:
```bash
9 vulnerabilities found - Packages audited: 2182
Severity: 9 Moderate
```

Number of packages reduced by ~750, `node_modules` size by ~400 MB

### CODE REVIEW

**Review by keshan3262 — DISMISSED** [review]

File: TODO.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+- Upgrade @taquito
+- Clean `yarn` `resolutions`
+- Is this error new? In console of running iOS build:
+```bash
+  Possible Unhandled Promise Rejection (id: 0):
+  Error: [messaging/unknown] The operation couldn’t be completed. No APNS token specified before fetching FCM Token
```
Comment (lendihop) [inline TODO.md:6]: These errors are related to firebase-messaging, which is used for push notifications, which don't work on a simulators. Does this errors appear on a real device?

**Review by lendihop — APPROVED** [review]

**Review by keshan3262 — APPROVED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Review by lendihop — APPROVED** [review]

**Review by lourenc — APPROVED** [review]
GJ 👍

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.16_936

**Review by NovakVlad2424 — APPROVED** [review]
ios done: 1.16_936

**Pull request merged**